backend/jupiterbrain.go

```diff
@@ -286,11 +299,19 @@ func (p *jupiterBrainProvider) Start(ctx gocontext.Context, startAttributes *Sta
 			}
 
 			if ip == nil {
-				time.Sleep(p.bootPollSleep)
+				select {
+				case <-time.After(p.bootPollSleep):
+				case <-ctx.Done():
+				}
+
 				continue
 			}
 
-			conn, err := net.Dial("tcp", fmt.Sprintf("%s:22", ip.String()))
+			dialer := &net.Dialer{
+				Timeout: time.Second,
```

Comment:
When I'm in doubt, I prefer to push it to a configurable value :grin:

That being said, one second timeout on dial seems low to me.  `3s`?  idk